odoo/fields.py

```diff
@@ -2792,6 +2795,93 @@ def _record_ids_per_res_model(self, records):
         return model_ids
 
 
+class Command(enum.IntEnum):
+    """
+    :class:`~odoo.fields.One2many` and :class:`~odoo.fields.Many2many` fields
+    expect a special command to manipulate the relation they implement.
+    Internally, each command is a tuple where the first element is a mandatory
+    integer that identifies the command, the two next elements are arguments
+    related to the command.
+
+    Please refer to the :meth:`~odoo.models.BaseModel.write` documentation for
+    full details on the behavior of each command.
+    """
+
+    CREATE = 0
+    UPDATE = 1
+    DELETE = 2
+    UNLINK = 3
+    LINK = 4
+    CLEAR = 5
+    SET = 6
+
+    @classmethod
+    def create(cls, values):
+        """
+        Create new records in the comodel using ``values``, link the created
+        records to ``self``.
+
+        In case of a **Many2many** relation, one unique new record is created
+        in the comodel such that all records in `self` are linked to the new
+        record.
+
+        In case of a **One2many** relation, one new record is created in the
+        comodel for every record in ``self`` such that every record in ``self``
+        is linked to exactly one of the new records.
+        """
+        return (cls.CREATE, 0, values)
```

Comment:
@rco-odoo should this support and / or document virtual ids management?

Reply:
Our opinion on the matter is that it is simple just to return a tuple triple using the constant. So I don't think we should support such ids.

Reply:
Managing virtual ids is only done in method `onchange` and in some x2many field methods.  Supporting this has no added value for business code.